### PR TITLE
#Added S3IO-Operations

### PR DESCRIPTION
**Pull request by aayushisharma88**
Signed-off-by: root <aayushi.sharma@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: comptests/prov/test_prov_k8s_cortx.py [hunk 1]
```diff
@@ -436,3 +440,32 @@ def test_32654(self):
...
+                    endpoint_url=client_config_res[2][2])
+        create_bucket_resp = s3t_obj.create_bucket(bucket_name)
+        LOGGER.info("Created bucket name %s", bucket_name)
+        assert_utils.assert_true(create_bucket_resp[0], create_bucket_resp[1])
+        bucket_resp = s3t_obj.bucket_list()
+        LOGGER.debug("bucket_list %s", bucket_resp[1])
+        cmd = commands.CMD_AWSCLI_HEAD_BUCKET.format(bucket_name) + " --endpoint-url " + client_config_res[2][2]
+        resp = system_utils.run_local_cmd(cmd=cmd, chk_stderr=True)
+        assert_utils.assert_true(resp[0], resp[1])
+        resp=s3t_obj.delete_bucket(bucket)
```
Comment (priyankaborawake) [inline comptests/prov/test_prov_k8s_cortx.py:467]: it should be bucket_name here.

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 2]
```diff
@@ -1965,3 +1981,18 @@ def update_res_limit_cortx(self, filepath):
...
+        returns access key and secrets key
+        """
+        with open(filepath) as soln:
+            conf = yaml.safe_load(soln)
+            parent_key = conf['solution']  # Parent key
+            common = parent_key['common']
+            access_key = common["s3"]["default_iam_users"]["auth_admin"]
+            secrets = parent_key['secrets']
+            secret_key = secrets["content"]["s3_auth_admin_secret"]
+        return access_key, secret_key
```
Comment (priyankaborawake) [inline libs/prov/prov_k8s_cortx_deploy.py:1998]: add new line here.

**Review by pragamj — APPROVED** [review]

File: comptests/prov/test_prov_k8s_cortx.py [hunk 3]
```diff
@@ -436,3 +440,32 @@ def test_32654(self):
...
+        bucket_resp = s3t_obj.bucket_list()
+        LOGGER.debug("bucket_list %s", bucket_resp[1])
+        cmd = commands.CMD_AWSCLI_HEAD_BUCKET.format(bucket_name) + " --endpoint-url " + client_config_res[2][2]
+        resp = system_utils.run_local_cmd(cmd=cmd, chk_stderr=True)
+        assert_utils.assert_true(resp[0], resp[1])
+        resp=s3t_obj.delete_bucket(bucket_name)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Uploading the bucket")
+        self.deploy_lc_obj.basic_io_write_read_validate(s3t_obj, bucket_name)
+
```
Comment (priyankaborawake) [inline comptests/prov/test_prov_k8s_cortx.py:471]: Add new line.

**Review by ravindrachoudhari — APPROVED** [review]

**Review by kachhwahadivya — APPROVED** [review]
Approving for merge

**Pull request merged**